pkg/validatorfactory/factory.go

```diff
@@ -271,60 +348,31 @@ func (s *ValidatorFactory) ValidatorsForGVK(gvk schema.GroupVersionKind) (*Valid
 
 	ssf := NewStructuralSchemaFactory(openapiSpec2.Components.Schemas)
 
+	namespaced := sets.New[schema.GroupVersionKind]()
+	if openapiSpec.Paths != nil {
+		for path, pathInfo := range openapiSpec.Paths.Paths {
+			for _, gvk := range getGVKsFromPath(pathInfo) {
+				if !namespaced.Has(gvk) {
+					if strings.Contains(path, "namespaces/{namespace}") {
+						namespaced.Insert(gvk)
```

Comment:
@apelisse do you think we should be worried here about a cluster-scoped resource and a namespace-scoped resource sharing the same kind?

Reply:
Not sure, maybe? It'd be nice if we could detect name conflicts though (i.e. two similar name in the same namespace vs two similar name in different namespaces).

Reply:
>  (i.e. two similar name in the same namespace vs two similar name in different namespaces).

Not sure how this applies here since kinds do not belong to namespaces

Reply:
Yeah, my statement was more general. Beside that, technically you could have a cluster-scoped and a namespace-scoped resource that use the same kind yes.